tests/components/unifiprotect/test_camera.py

```diff
@@ -0,0 +1,358 @@
+"""Test the UniFi Protect camera platform."""
+from __future__ import annotations
+
+from copy import copy
+from typing import cast
+from unittest.mock import AsyncMock, Mock
+
+from pyunifiprotect.data import Camera as ProtectCamera
+from pyunifiprotect.exceptions import NvrError
+
+from homeassistant.components.camera import Camera, async_get_image
+from homeassistant.components.unifiprotect.const import (
+    ATTR_BITRATE,
+    ATTR_CHANNEL_ID,
+    ATTR_FPS,
+    ATTR_HEIGHT,
+    ATTR_WIDTH,
+    DEFAULT_ATTRIBUTION,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.components.unifiprotect.data import ProtectData
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from .conftest import MockEntityFixture, time_changed
+
+
+async def validate_camera_entity(
+    hass: HomeAssistant,
+    camera: ProtectCamera,
+    channel_id: int,
+    secure: bool,
+    rtsp_enabled: bool,
+    enabled: bool,
+):
+    """Validate a camera entity."""
+
+    channel = camera.channels[channel_id]
+
+    entity_name = f"{camera.name} {channel.name}"
+    unique_id = f"{camera.id}_{channel.id}"
+    if not secure:
+        entity_name += " Insecure"
+        unique_id += "_insecure"
+    entity_id = f"camera.{entity_name.replace(' ', '_').lower()}"
+
+    entity_registry = er.async_get(hass)
+    entity = entity_registry.async_get(entity_id)
+    assert entity
+    assert entity.disabled is (not enabled)
+    assert entity.unique_id == unique_id
+
+    if not enabled:
```

Comment:
Please don't include conditional logic in the tests. Each test should just have one branch. Use pytest.mark.parametrize to reuse test logic.

Reply:
Not trying to argue this, because I would normally agree, but the branching logic here is no coming from code or anything that is indeterministic. This is just a re-usable helper method used below by the setup test to verify initial entity state. 

All of the branches are hardcoded within that test. If a single Camera device is setup with all 3 RTSP channels enabled, it will generate 6 entities, 1 that is enabled and 5 that are disabled. That cannot really be parametrized with pytest. Re-writing this without if cases would create a lot more code and make it much less readable/easy to follow.

Reply:
I would make it so there is a parameter to verify each entity in its own test, but that would mean initializing hass 9 times (increase in test runtime) instead of 1 and I could not test setting up multiple camera entities at once.

Reply:
It's the opposite. It's more readable to avoid branching. Readable code doesn't mean shorter more convoluted code.

It looks like we could parametrize the test somehow at least by changing the way we assert below. Ie make the assert result a parameter. We don't need to use pytest.mark.parametrize if we want to test all entities in the same test but we can still avoid condition logic.